libevmasm/KnownState.cpp

```diff
@@ -154,8 +154,10 @@ KnownState::StoreOperation KnownState::feedItem(AssemblyItem const& _item, bool
 				);
 				break;
 			default:
-				bool invMem = SemanticInformation::invalidatesMemory(_item.instruction());
-				bool invStor = SemanticInformation::invalidatesStorage(_item.instruction());
+				bool invMem =
+					SemanticInformation::memory(_item.instruction()) == yul::SideEffects::Write;
```

Comment:
TODO, perhaps it should actually bye `SideEffects::Read` for memory. Because `keccak` and `mload` can mess with `msize()`